modules/javafx.graphics/src/main/java/javafx/scene/Scene.java

```diff
@@ -1301,11 +1298,55 @@ static WritableImage doSnapshot(Scene scene,
             height = (int)wimg.getHeight();
         }
 
+        int maxTextureSize = PrismSettings.maxTextureSize;
+        if (height > maxTextureSize || width > maxTextureSize) {
+            // The requested size for the screenshot is to big to fit a single texture,
+            // so we need to take several snapshot tiles and merge them into single image (fixes JDK-8088198)
+            int verticalTileNb = (int) Math.ceil(height / (double) maxTextureSize);
+            int horizontalTileNb = (int) Math.ceil(width / (double) maxTextureSize);
```

Comment:
I think the proposed code changes are wrong in case that `height / maxTextureSize` is an exact integer. (Same for width). I normally add the 1 only if `height  % maxTextureSize != 0`

Reply:
Indeed, substituting the floating math for integer math requires an extra test on the remainder to be correct. 
Since this code isn't located in a tight loop, I'm not sure this optimization is worth making the code less straight forward. What do you all think?

Reply:
I think it is not worth it.

Reply:
I agree. I do like the suggestion to rename the variables, though.

Reply:
done in 501917284e0490d16b1831fcd854e31a779449b9